dev/com.ibm.ws.ejbcontainer.timer.np.config_fat/test-applications/NpTimerConfigRetryWeb.war/src/com/ibm/ws/ejbcontainer/timer/np/config/retry/web/NpTimerConfigRetryServlet.java

```diff
@@ -616,7 +629,13 @@ public void testOverlappingRetriesAndRegularlyScheduled() throws Exception {
         ArrayList<Long> timestamps = (ArrayList<Long>) props.get(TimerRetryDriverBean.TIMESTAMP_KEY);
         @SuppressWarnings("unchecked")
         ArrayList<Long> nextTimes = (ArrayList<Long>) props.get(TimerRetryDriverBean.NEXTTIMEOUT_KEY);
-        assertEquals("Attempt count **" + count + "** was not the expected value of 7.", 7, count);
+
+        // Tolerate a slow system; if time to trip latch is > 22, then an 8th attempt for a makeup may occur
+        if (System.currentTimeMillis() - startTime < 22) {
```

Comment:
It can be better to use System.nanoTime() which is not vulnerable to system clock changes.

Reply:
updated to nanoTime